### DISCUSSION TITLE
stroke and strokewidth not working with Pango

### DISCUSSION THREAD

**Discussion by srini8080** [ImageMagick/ImageMagick#4726] [Digital Image Processing]
convert F5.png \( +size -background transparent -pointsize 100 -stroke red -strokewidth 50 -size 850x400 -define pango:align="center"  pango:"<span font_desc='Arial' foreground='black' font_weight='heavy'>"THIS IS A TEST"</span>" \) -gravity South -geometry +0+130 -compose over -composite F6.png

Stroke and Strokewidth are not working with Pango. Anything I am doing wrong?

Thanks,

**Comment by urban-warrior** [comment]
Use the Pango markup language, see https://legacy.imagemagick.org/Usage/text/#pango_markup.

**Reply by srini8080** [reply]
I did not find stroke and strokewidth as a part of Pango markup

**Comment by fearocanity** [comment]
Have you tried `-morphology EdgeOut Disk:"3"`, apply this on the back of the pango with black foreground, and composite it with the same pango but in white foreground.

Example:
```
convert -size 800x -background none -gravity center -morphology EdgeOut Disk:"3" -fill black pango:"<span size='35pt' foreground='black'>THIS IS A TEST</span>" -background none -fill white pango:"<span size='35pt' foreground='white'>THIS IS A TEST</span>" -composite test.png
```

![test](https://github.com/user-attachments/assets/cab0d2d0-517f-413a-aa70-b15a698044d0)

**Reply by srini8080** [reply]
excellent method, thanks
